File: Exemplos SmartPOS/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_SmartPOS_Flutter-font/lib/IntentDigitalHubService/TERMICA/Commands/impressao_codigo_barras.dart
import '../termica_command.dart';

class ImpressaoCodigoBarras extends TermicaCommand {
  final int _tipo;
  final String _dados;
  final int _altura;
  final int _largura;
  final int _HRI;

  ImpressaoCodigoBarras(
      this._tipo, this._dados, this._altura, this._largura, this._HRI)
      : super('ImpressaoCodigoBarras');

  @override
  get functionParametersJson => {
        'tipo': _tipo,
        'dados': _dados,
        'altura': _altura,
        'largura': _largura,
        'HRI': _HRI,
      };
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_SmartPOS_Ionic-font/src/intentDigitalHubService/TERMICA/Commands/ImpressaoCodigoBarras.ts
import {TermicaCommand} from '../TermicaCommand';

export class ImpressaoCodigoBarras extends TermicaCommand {
  readonly tipo: number;
  readonly dados: string;
  readonly altura: number;
  readonly largura: number;
  readonly HRI: number;

  constructor(
    tipo: number,
    dados: string,
    altura: number,
    largura: number,
    HRI: number,
  ) {
    super('ImpressaoCodigoBarras');
    this.tipo = tipo;
    this.dados = dados;
    this.altura = altura;
    this.largura = largura;
    this.HRI = HRI;
  }

  functionParameters(): object {
    return {
      tipo: this.tipo,
      dados: this.dados,
      altura: this.altura,
      largura: this.largura,
      HRI: this.HRI,
    };
  }
}
